nullability/v0.2/nullability-v0.2.graphql

```diff
@@ -0,0 +1,100 @@
+"""
+Indicates that a position is semantically non null: it is only null if there is a matching error in the `errors` array.
+In all other cases, the position is non-null.
+
+Tools doing code generation may use this information to generate the position as non-null:
+
+```graphql
+type User {
+    # email is semantically non-null and can be generated as non-null by error-handling clients.
+    email: String @semanticNonNull
+}
+```
+
+The `field` argument is the name of the field if `@semanticNonNull` is applied to an object type extension:
+
+```graphql
+# extend the base schema to make User.email semantically non-null.
+extend type User @semanticNonNull(field: "email")
+```
+
+`field` is used for client applications that do not control the base schema and must use type extensions.
+When used on a field definition, `field` must be null.
+
+The `level` argument indicates what level is semantically non null in case of lists:
+
+```graphql
+type User {
+    # friends is semantically non null
+    friends: [User] @semanticNonNull # same as @semanticNonNull(level: 0)
+
+    # friends[0], ... are semantically non null
+    friends: [User] @semanticNonNull(level: 1)
+
+    # friends, friends[0], ... are all semantically non null
+    friends: [User] @semanticNonNull(level: 0) @semanticNonNull(level: 1)
+}
+```
+
+`level` is zero indexed.
+Passing a negative level or a level greater than the list dimension is an error.
+
+"""
+directive @semanticNonNull(field: String = null, level: Int = 0) repeatable on FIELD_DEFINITION | OBJECT
```

Comment:
Another aside: If we support the ability to define `@semanticNonNull` on an object, presumably we also need to support it on an interface?

Reply:
Excellent point 👍 , will edit.